src/server/services/home.ts

```diff
@@ -18,47 +18,44 @@ import { AuthProfileAndToken } from '../auth/auth-types';
 async function fetchPrivate(
   requestID: RequestID,
   authProfileAndToken: AuthProfileAndToken
-) {
+): Promise<ApiResponse<BAGData[] | null>> {
   const BRP = await fetchBRP(requestID, authProfileAndToken);
 
-  let MY_LOCATION: ApiResponse<(BAGData | null)[] | null>;
-
-  if (BRP.status === 'OK' && isMokum(BRP.content)) {
-    const location = (await fetchBAG(requestID, BRP.content.adres))?.content;
+  if (BRP.status === 'OK') {
+    if (isMokum(BRP.content)) {
+      const location = (await fetchBAG(requestID, BRP.content.adres))?.content;
 
-    // No BAG location found
-    if (!location?.latlng) {
-      MY_LOCATION = apiSuccessResult([
-        {
-          latlng: {
-            lat: DEFAULT_LAT,
-            lng: DEFAULT_LNG,
+      // No BAG location found
+      if (!location?.latlng) {
+        return apiSuccessResult([
+          {
+            latlng: {
+              lat: DEFAULT_LAT,
+              lng: DEFAULT_LNG,
+            },
+            address: null,
+            profileType: 'private',
           },
+        ]);
+      } else {
+        // BAG Location found!
+        return apiSuccessResult([
+          Object.assign(location, { profileType: 'private' }),
+        ]);
+      }
+    } else {
+      // Not a Mokum address
+      return apiSuccessResult([
+        {
+          latlng: null,
           address: null,
           profileType: 'private',
         },
       ]);
-    } else {
-      // BAG Location found!
-      MY_LOCATION = apiSuccessResult([
-        Object.assign(location, { profileType: 'private' }),
-      ]);
     }
-  } else if (BRP.status === 'OK' && !isMokum(BRP.content)) {
-    // Not a Mokum address
-    MY_LOCATION = apiSuccessResult([
-      {
-        latlng: null,
-        address: null,
-        profileType: 'private',
-      },
-    ]);
-  } else {
-    // Proper data concerning a Mokum location not found
-    MY_LOCATION = apiDependencyError({ BRP });
   }
 
-  return MY_LOCATION;
```

Comment:
```suggestion
  if (BRP.status !== 'OK') {
    return apiSuccessResult([
      {
        latlng: null,
        address: null,
        profileType: 'private',
      },
    ]);
  }

  if (isMokum(BRP.content)) {
    const location = (await fetchBAG(requestID, BRP.content.adres))?.content;

    if (location?.latlng) {
      // BAG Location found!
      return apiSuccessResult([
        Object.assign(location, { profileType: 'private' }),
      ]);
    }
    // No BAG location found
    return apiSuccessResult([
      {
        latlng: {
          lat: DEFAULT_LAT,
          lng: DEFAULT_LNG,
        },
        address: null,
        profileType: 'private',
      },
    ]);
  }
```

Reply:
Als er een paar dingetjes kwa logica omgedraaid wordt dan wordt het wat simpler volgens mij.

Reply:
Heb alleen een beetje gefaalt het stuk goed te highlighten denk ik... Had ook de deletions meegenomen, of is dat juist goed? hehe

Reply:
Ik snap de diff niet idd.

Paste anders de functie zoals je hem zou versimpelen

Reply:
Ja snap ik, alright dit is de hele functie:
```
async function fetchPrivate(
  requestID: RequestID,
  authProfileAndToken: AuthProfileAndToken
): Promise<ApiResponse<BAGData[] | null>> {
  const BRP = await fetchBRP(requestID, authProfileAndToken);

  if (BRP.status !== 'OK') {
    return apiSuccessResult([
      {
        latlng: null,
        address: null,
        profileType: 'private',
      },
    ]);
  }
  if (isMokum(BRP.content)) {
    const location = (await fetchBAG(requestID, BRP.content.adres))?.content;
    if (location?.latlng) {
      // BAG Location found!
      return apiSuccessResult([
        Object.assign(location, { profileType: 'private' }),
      ]);
    }
    // No BAG location found
    return apiSuccessResult([
      {
        latlng: {
          lat: DEFAULT_LAT,
          lng: DEFAULT_LNG,
        },
        address: null,
        profileType: 'private',
      },
    ]);
  }

  return apiDependencyError({ BRP });
}
```

Reply:
Ah, ik zie 'm. Het is alleen niet dezelfde functionaliteit.